src/serial.rs

```diff
@@ -36,24 +23,20 @@ use crate::gpio::gpioc::{PC10, PC11, PC4, PC5};
     feature = "stm32f358",
     feature = "stm32f398"
 ))]
-use crate::gpio::gpiod::{PD5, PD6, PD8, PD9};
-#[cfg(any(
-    feature = "stm32f302",
-    feature = "stm32f303xb",
-    feature = "stm32f303xc",
-    feature = "stm32f303xd",
-    feature = "stm32f303xe",
-    feature = "stm32f373",
-    feature = "stm32f378",
-    feature = "stm32f328",
-    feature = "stm32f358",
-    feature = "stm32f398"
-))]
-use crate::gpio::gpioe::{PE0, PE1, PE15};
+use crate::gpio::{gpiod, gpioe};
+
+#[cfg(feature = "stm32f303")]
+mod dma_imports {
+    pub use crate::dma;
+    pub use as_slice::{AsMutSlice, AsSlice};
+    pub use core::{
+        ops::{Deref, DerefMut},
+        pin::Pin,
+    };
+}
 
-use crate::gpio::AF7;
-use crate::rcc::{Clocks, APB1, APB2};
-use crate::time::Bps;
+#[cfg(feature = "stm32f303")]
+use dma_imports::*;
```

Comment:
I cleaned up the imports a bit while I added my own ones. I noticed that model-specific imports add a lot of duplicated `cfg` statements. It would help readability a lot if we could reduce this duplication.

According to [this SO answer](https://stackoverflow.com/questions/43070430/is-there-a-way-to-use-the-cfgfeature-check-on-multiple-statements), we can solve this two ways:

1. Using an ad-hoc submodule and then re-exporting its members in the parent. This is the technique you see here.
2. Using the [cfg-if](https://crates.io/crates/cfg-if) crate.

I'm not too fond of the submodule solution, as it still adds unnecessary visual noise and still needs the `cfg` to be repeated twice. The `cfg-if` solution looks better to me, but would require an additional dependency.

Reply:
I welcome this change :+1: